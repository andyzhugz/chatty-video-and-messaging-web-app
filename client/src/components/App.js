import { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Login from './Login';
import Register from './Register';
import MainPage from './MainPage';
import Profile from './Profile';
import Status from './Status';
import ForgetPassword from './ForgetPassword';
import OtherProfile from './OtherProfile';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/">
            <Login />
          </Route>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/register">
            <Register />
          </Route>
          <Route exact path="/profile">
            <Profile />
          </Route>
          <Route exact path="/main">
            <MainPage />
          </Route>
          <Route exact path="/status">
            <Status />
          </Route>
          <Route exact path="/changePassword">
            <ForgetPassword />
          </Route>
          <Route exact path="/otherProfile/:otherUser">
            <OtherProfile />
          </Route>
        </div>
      </Router>
    );
  }
}

export default App;
