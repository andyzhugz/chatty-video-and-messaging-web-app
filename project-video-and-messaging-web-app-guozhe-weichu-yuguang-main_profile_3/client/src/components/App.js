import { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Login from './login';
import Register from './Register';
import MainPage from './MainPage';
import Profile from './Profile';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <Route exact path="/">
            <Login />
          </Route>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/register">
            <Register />
          </Route>
          <Route exact path="/profile">
            <Profile />
          </Route>
          <Route exact path="/main">
            <MainPage />
          </Route>
        </div>
      </Router>
    );
  }
}

export default App;
