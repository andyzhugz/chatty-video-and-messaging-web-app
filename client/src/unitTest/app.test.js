import React from 'react';
import App from '../components/App.js';

const renderer = require('react-test-renderer');

// snapshot testings
describe('Test snapshot', () => {
  test('Test Profile', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});