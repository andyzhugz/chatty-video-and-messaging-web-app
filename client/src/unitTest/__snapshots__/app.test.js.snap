// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Test Profile 1`] = `
<div
  className="App"
>
  <div>
    <h1
      align="center"
    >
      Welcome to Our App
    </h1>
    <div
      className="main"
    >
      <p
        align="center"
        className="sign"
      >
        Login
      </p>
      <form
        className="formLogin"
      >
        <input
          align="center"
          className="userName"
          id="userName"
          placeholder="Email: cat@gmail.com"
          type="text"
        />
        <input
          align="center"
          className="pass"
          id="password"
          placeholder="Password"
          type="password"
        />
        <a
          href="/changePassword"
          onClick={[Function]}
        >
          <p
            align="center"
            className="forgotPwd"
          >
            Forgot Password?
          </p>
        </a>
        <input
          className="loginBtn"
          disabled={false}
          id="loginBtn"
          onClick={[Function]}
          type="button"
          value="Sign In"
        />
        <a
          href="/Register"
          onClick={[Function]}
        >
          <button
            align="center"
            className="registerBtn"
            id="registerBtn"
          >
            <b>
              Register
            </b>
          </button>
        </a>
        <br />
      </form>
    </div>
  </div>
</div>
`;
